src/figuresClasses.ts

```diff
@@ -1,11 +1,86 @@
-export interface Figure {}
+type ValidShapes = 'triangle' | 'circle' | 'rectangle';
+type ValidColors = 'red' | 'green' | 'blue';
 
-export class Triangle implements Figure {}
+export interface Figure {
+  shape: ValidShapes;
+  color: ValidColors;
+  getArea: () => number;
+}
+
+export class Triangle implements Figure {
+  public shape: ValidShapes = 'triangle';
+
+  constructor(
+    public color: ValidColors,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    const sides = [a, b, c];
+    const longestSide = Math.max(...sides);
+    let sumOfOtherSides = 0;
+
+    for (const side of sides) {
+      if (side < longestSide) {
+        sumOfOtherSides += side;
+      }
+    }
```

Comment:
do u need `for loop` for this?
```suggestion
    const sumOfOtherSides = (a + b + c) - longestSide;
```